sql/multiregion: move telemetry names to break catpb->tree dep

ajwerner opened the pull request:
It was a bottleneck in the build graph. See individual commits.

Release note: None

ajwerner committed:
sql/multiregion: move telemetry names to break catpb->tree dep

It was a bottleneck in the build graph.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/82432)
<!-- Reviewable:end -->

ajwerner committed:
sql/privilege,sql/sem/tree: move TargetObjectType to privilege

This is required to break the dependency from catpb on tree. Doing so should
improve build times and generally provide a bit more cleanliness in the
dependency graph.

Release note: None

chengxiong-ruan reviewed (approved):
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/82432#-:-N3twxqY8AYG39hq50Qc:b-uwkowd)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @RichardJCai)



<!-- Sent from Reviewable.io -->

RichardJCai reviewed (approved):
Looked at privilege commit, LGTM

ajwerner commented:
TFTR!

bors r+
